components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/dao/SessionInfoDAO.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authenticator.oidc.LogoutServerException;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.AnalyticsAttributes.SESSION_ID;
+import static org.wso2.carbon.identity.application.mgt.ApplicationConstants.IDP_NAME;
+
+public class SessionInfoDAO {
+
+    private static final Log log = LogFactory.getLog(SessionInfoDAO.class);
+
+    /**
+     * Retrieve the session details of a given OIDC sid from the database.
+     *
+     * @param oidcSId Session Id of the OIDC Logout Request.
+     * @return Map of session details.
+     * @throws LogoutServerException If DB execution fails.
+     */
+    public Map<String, String> getSessionDetails(String oidcSId) throws LogoutServerException {
```

Comment:
This DAO seems to required in two places as of now. What about moving this to the framework?
@piraveena

Reply:
yes we can move this to framework and refactor saml as well

Reply:
can we rename ths oidcSId ->fedIdpSessionId as well

Reply:
Moved the function to UserSessionDAO https://github.com/wso2/carbon-identity-framework/pull/3314